aom/src/main/java/com/nedap/archie/rminfo/MetaModel.java

```diff
@@ -0,0 +1,314 @@
+
+    public MetaModel(ModelInfoLookup selectedModel, BmmModel selectedBmmModel, AomProfile selectedAomProfile) {
+        this.selectedModel = selectedModel;
+        this.selectedBmmModel = selectedBmmModel;
+        this.selectedAomProfile = selectedAomProfile;
+    }
+
+    public ModelInfoLookup getSelectedModel() {
+        return selectedModel;
+    }
+
+    public BmmModel getSelectedBmmModel() {
+        return selectedBmmModel;
+    }
+
+    public AomProfile getSelectedAomProfile() {
+        return selectedAomProfile;
+    }
+
+    @Override
+    public boolean isMultiple(String typeName, String attributeName) {
+        if(getSelectedBmmModel() != null) {
+            BmmClass classDefinition = getSelectedBmmModel().getClassDefinition(BmmDefinitions.typeNameToClassKey(typeName));
+            if (classDefinition != null) {
+                //TODO: don't flatten on request, create a flattened properties cache just like the eiffel code for much better performance
+                BmmClass flatClassDefinition = classDefinition.flattenBmmClass();
+                BmmProperty bmmProperty = flatClassDefinition.getProperties().get(attributeName);
+                return isMultiple(bmmProperty);
+            }
+        } else {
+            RMAttributeInfo attributeInfo = selectedModel.getAttributeInfo(typeName, attributeName);
+            return attributeInfo != null && attributeInfo.isMultipleValued();
+        }
+        return false;
+    }
+
+    private boolean isMultiple(BmmProperty bmmProperty) {
+        if(bmmProperty == null) {
+            return false;
+        } else if(bmmProperty instanceof BmmContainerProperty) {
+            return bmmProperty != null && ((BmmContainerProperty) bmmProperty).getCardinality().has(2);
+        } else {
+            return false;
+        }
+    }
+
+    @Override
+    public boolean rmTypesConformant(String childTypeName, String parentTypeName) {
+        if(getSelectedBmmModel() != null) {
+            BmmModel selectedBmmModel = getSelectedBmmModel();
+            String parentClassName = BmmDefinitions.typeNameToClassKey(parentTypeName);//generics stripped
+            String childClassName = BmmDefinitions.typeNameToClassKey(childTypeName);//generics stripped
+            //TODO: generics as well. get the array and check each type?
+            BmmClass childClass = selectedBmmModel.getClassDefinition(childClassName);
+            if(childClass == null) {
+                return true;//will be checked elsewhere
+            }
+            List<String> allAncestors = childClass.findAllAncestors();
+            allAncestors = allAncestors.stream().map((s) -> s.toUpperCase()).collect(Collectors.toList());
+            if(!parentClassName.equalsIgnoreCase(childClassName) && !allAncestors.contains(parentClassName)) {
+                return false;
+            }
+            return true;
+        } else {
+            return selectedModel.rmTypesConformant(childTypeName, parentTypeName);
```

Comment:
Since both models can be null at the moment, this might throw a NullPointerException. Same thing for other methods here